### ISSUE TITLE
ThisAssembly.Git exception when project has Git submodule

### ISSUE THREAD

**Issue by TibbsTerry** [devlooped/ThisAssembly#303]
## Describe the Bug

ThisAssembly.Git produces a build error when solution has Git submodules

ThisAssembly.Git.targets(71,7): error MSB4184: The expression "[System.IO.File]::ReadAllText(/path/;/path/anotherpath/.git/HEAD)" cannot be evaluated. Could not find a part of the path '/path/;/path/anotherpath/.git/HEAD'

Issue is in ThisAssembly.Git.targets line 71  `$([System.IO.File]::ReadAllText`

    <PropertyGroup Condition="'$(RepositoryBranch)' == '' and '$(RepositoryRoot)' != ''">
      
      <RepositoryHead>$(RepositoryRoot).git/HEAD</RepositoryHead>
      <RepositoryBranch Condition="Exists($(RepositoryHead))">$([System.IO.File]::ReadAllText($(RepositoryHead)).Trim())</RepositoryBranch>
      <RepositoryBranch Condition="$(RepositoryBranch) != ''">$(RepositoryBranch.TrimStart("ref:").Trim().TrimStart("refs").Trim("/").TrimStart("heads").Trim("/").Trim())</RepositoryBranch>
    </PropertyGroup>

I believe because `@(SourceRoot)` will also return submodule information

**Comment by kzu** [comment]
Hm. I wonder how come `Exists($(RepositoryHead))` would return true and then `File.ReadAllText` fail? 🤔

**Comment by TibbsTerry** [comment]
Fix in 
ThisAssembly.Git.targets

  ```
<ItemGroup>
    <_ThisAssemblyGitSourceRoot Include="@(SourceRoot -> WithMetadataValue('SourceControl', 'git'))" 
                                Exclude="@(SourceRoot -> HasMetadata('NestedRoot'))" />
</ItemGroup>
```

**Comment by TibbsTerry** [comment]
Also think the build failure may be just a Linux issue. Anyway needs the change above, so that the Branch constant is set.

**Comment by kzu** [comment]
I believe this is fixed by your PR #334

**Comment by kzu** [comment]
BTW, just shipped it as 1.5.0-beta.
